inginious/frontend/plugins/task_hints/pages/user_hint_manager.py

```diff
@@ -0,0 +1,135 @@
+from inginious.frontend.parsable_text import ParsableText
+
+
+class UserHintManager(object):
+    """
+        Manage the user's hints data in database. This includes the hints
+        that were unlocked, the individual penalty, and cumulative
+        penalty, to be applied in the student's final grade for the
+        respective task.
+    """
+
+    def __init__(self, username, task_id, database):
+
+        self._username = username
+        self._task_id = task_id
+        self._database = database
+
+    def get_hint_content_by_status(self, hints):
+
+        """
+            This is a method to check each hint unlocked status, and return the left content
+            if it is unlocked. Also set a 'True' value in the 'unlocked'
+            attribute for the unlocked hints to show their additional data
+            in the modal hints in the task.
+        """
+        user_hints = self._database.user_hints.find_one({"taskid": self._task_id,
+                                                         "username": self._username})
+        if user_hints is None:
+            user_hints = self.insert_default_user_hints()
+            return 200, ""
+
+        self.update_unlocked_user_hints(user_hints, hints)
+
+        hints_to_show = {}
+        unlocked_hints = user_hints["unlocked_hints"]
```

Comment:
So you first update the unlocked_hints, but then you access the `unlocked_hints` from an outdated `user_hints` value, as the query to DB was done before. Shouldn't it be a new query done? or return the new unlocked_hints in the `update_unlocked_user_hints` method.

Reply:
Yes, it was changed because hints were not loading when 'user_hints' was added to the database.